packages/plugins/jwt/src/plugin.ts

```diff
@@ -0,0 +1,242 @@
+          );
+
+          throw Error(`Authentication is not available at the moment.`);
+        }
+
+        // Verify the token with the signing key.
+        const verified = await verify(
+          logger,
+          token,
+          signingKey,
+          normalizedOptions.tokenVerification,
+        );
+
+        if (!verified) {
+          throw unauthorizedError(`Unauthenticated`);
+        }
+
+        if (verified) {
+          // Link the verified payload with the request (see `onContextBuilding` for the reading part)
+          payloadByRequest.set(payload.request, verified);
+        }
+      } catch (e) {
+        // User-facing errors should be handled based on the configuration.
+        // These errors are handled based on the value of "reject.invalidToken" config.
+        if (e instanceof GraphQLError) {
+          if (normalizedOptions.reject.invalidToken) {
+            throw e;
+          }
+
+          return;
+        }
+
+        // Server/internal errors should be thrown, so they can be handled by the error handler and be masked.
+        throw e;
+      }
+    },
+    onContextBuilding({ context, extendContext }) {
+      if (context.request == null) {
+        throw new Error(
+          'Request is not available on context! Make sure you use this plugin with GraphQL Yoga.',
+        );
+      }
+
+      // Get the payload and inject it into the GraphQL context.
+      const payload = payloadByRequest.get(context.request);
+      if (payload !== null && normalizedOptions.extendContextFieldName) {
+        extendContext({
+          [normalizedOptions.extendContextFieldName]: payload,
+        });
+      }
+    },
+  };
+}
+
+function badRequestError(message: string, options?: Parameters<typeof createGraphQLError>[1]) {
+  return createGraphQLError(message, {
+    extensions: {
+      http: {
+        status: 400,
+      },
+    },
+    ...options,
+  });
+}
+
+function unauthorizedError(message: string, options?: Parameters<typeof createGraphQLError>[1]) {
+  return createGraphQLError(message, {
+    extensions: {
+      http: {
+        status: 401,
+      },
+    },
+    ...options,
+  });
+}
+
+function verify(
+  logger: YogaLogger,
+  token: string,
+  signingKey: string,
+  options: VerifyOptions | undefined,
+) {
+  return new Promise((resolve, reject) => {
+    jsonwebtoken.verify(token, signingKey, options, (err, result) => {
+      if (err) {
+        logger.warn(`Failed to verify authentication token: `, err);
+        reject(unauthorizedError('Unauthenticated'));
+      } else {
+        resolve(result as JwtPayload);
+      }
+    });
+  });
+}
+
+type SigningKeyProvider = (kid?: string) => Promise<string>;
+
+function buildProvider(source: JwksSource): SigningKeyProvider {
+  switch (source.type) {
+    case 'inline': {
+      return () => Promise.resolve(source.signingKey);
```

Comment:
The promise is mandatory here ? From what I see, we just `await` it, and it is possible to `await` a string

Reply:
Fixed